st2actions/tests/unit/test_mistral_querier_v2.py

```diff
@@ -230,3 +240,78 @@ def test_query(self):
 
         self.assertEqual(action_constants.LIVEACTION_STATUS_SUCCEEDED, status)
         self.assertDictEqual(expected, result)
+
+    @mock.patch.object(
+        executions.ExecutionManager, 'get',
+        mock.MagicMock(side_effect=[
+            requests.exceptions.ConnectionError(),
+            MOCK_WF_EX]))
+    @mock.patch.object(
+        tasks.TaskManager, 'list',
+        mock.MagicMock(return_value=MOCK_WF_EX_TASKS))
+    @mock.patch.object(
+        action_service, 'is_action_canceled_or_canceling',
+        mock.MagicMock(return_value=False))
+    def test_query_get_workflow_retry(self):
+        (status, result) = self.querier.query(uuid.uuid4().hex, MOCK_QRY_CONTEXT)
+
+        expected = {
+            'k1': 'v1',
+            'tasks': copy.deepcopy(MOCK_WF_EX_TASKS_DATA),
+            'extra': {
+                'state': MOCK_WF_EX.state,
+                'state_info': MOCK_WF_EX.state_info
+            }
+        }
+
+        for task in expected['tasks']:
+            task['input'] = json.loads(task['input'])
+            task['result'] = json.loads(task['result'])
+            task['published'] = json.loads(task['published'])
+
+        self.assertEqual(action_constants.LIVEACTION_STATUS_SUCCEEDED, status)
```

Comment:
Do we have any test case where a maximum number of retries is reached and it still fails?